### PR TITLE
expose STL iterator interface for FairMQParts

### LINKED ISSUE #556: FairMQParts STL iterator API
It would be a nice to have a stl iterator API for FairMQParts, so that one can do:

```
for (auto && part : parts) {
  char * data = part->GetData();
  ...
}
```

### PR DESCRIPTION
**Pull request by dennisklein**
resolves #556

### CODE REVIEW

**Comment by dennisklein** [comment]
duplicate of #557 :(

**Comment by rbx** [comment]
keep this one, it's more complete with the tests :)

**Comment by dennisklein** [comment]
rebased to new `dev`

**Comment by dennisklein** [comment]
And there comes the mac again...

File: fairmq/test/parts/_iterator_interface.cxx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    FairMQParts mParts;
+    FairMQTransportFactoryZMQ mFactory;
+
+    RandomAccessIterator()
+    : mParts(FairMQParts{}),
+      mFactory(FairMQTransportFactoryZMQ{})
+    {
+        auto s1 = string{"1"};
+        auto s2 = string{"2"};
+        auto s3 = string{"3"};
```
Comment (rbx) [inline fairmq/test/parts/_iterator_interface.cxx]: These three guys go out of scope, that's why the test fails sometimes. make them members, then its fine.

Comment (dennisklein) [inline fairmq/test/parts/_iterator_interface.cxx]: Yes, thinking about it, putting stack data into a message is almost always a bad idea.

Comment (rbx) [inline fairmq/test/parts/_iterator_interface.cxx]: ah, but that needs a device... nevermind, fixing the test is simpler.

Comment (dennisklein) [inline fairmq/test/parts/_iterator_interface.cxx]: thx for spotting this, I changed it.

**Pull request merged**